pennylane_lightning/src/benchmarks/Bench_LinearAlgebra.cpp

```diff
@@ -0,0 +1,360 @@
+        benchmark::DoNotOptimize(vec2[sz - 1]);
+    }
+}
+BENCHMARK(omp_matrixVecProd_cmplx<float>)
+    ->RangeMultiplier(1l << 2)
+    ->Range(1l << 4, 1l << 8);
+
+BENCHMARK(omp_matrixVecProd_cmplx<double>)
+    ->RangeMultiplier(1l << 2)
+    ->Range(1l << 4, 1l << 8);
+
+#if __has_include(<cblas.h>) && defined _ENABLE_BLAS
+template <class T>
+static void blas_matrixVecProd_cmplx(benchmark::State &state) {
+    std::random_device rd;
+    std::mt19937_64 eng(rd());
+    std::uniform_real_distribution<T> distr;
+    auto sz = static_cast<size_t>(state.range(0));
+
+    std::vector<std::complex<T>> mat;
+    for (size_t i = 0; i < sz * sz; i++)
+        mat.push_back({distr(eng), distr(eng)});
+
+    std::vector<std::complex<T>> vec1;
+    for (size_t i = 0; i < sz; i++)
+        vec1.push_back({distr(eng), distr(eng)});
+
+    const auto tr = static_cast<CBLAS_TRANSPOSE>(Trans::NoTranspose);
+    constexpr std::complex<T> co{1, 0};
+    constexpr std::complex<T> cz{0, 0};
+
+    for (auto _ : state) {
+        std::vector<std::complex<T>> vec2(sz);
+
+        if constexpr (std::is_same_v<T, float>) {
+            cblas_cgemv(CblasRowMajor, tr, sz, sz, &co, mat.data(), sz,
+                        vec1.data(), 1, &cz, vec2.data(), 1);
+        } else if constexpr (std::is_same_v<T, double>) {
+            cblas_zgemv(CblasRowMajor, tr, sz, sz, &co, mat.data(), sz,
+                        vec1.data(), 1, &cz, vec2.data(), 1);
+        }
+
+        benchmark::DoNotOptimize(vec2[sz - 1]);
+    }
+}
+BENCHMARK(blas_matrixVecProd_cmplx<float>)
+    ->RangeMultiplier(1l << 2)
+    ->Range(1l << 4, 1l << 8);
+
+BENCHMARK(blas_matrixVecProd_cmplx<double>)
+    ->RangeMultiplier(1l << 2)
+    ->Range(1l << 4, 1l << 8);
+#endif
+
+template <class T>
+static void omp_matrixMatProd_cmplx(benchmark::State &state) {
+    std::random_device rd;
+    std::mt19937_64 eng(rd());
+    std::uniform_real_distribution<T> distr;
+    auto sz = static_cast<size_t>(state.range(0));
+
+    std::vector<std::complex<T>> m_left;
+    for (size_t i = 0; i < sz * sz; i++)
+        m_left.push_back({distr(eng), distr(eng)});
+
+    std::vector<std::complex<T>> m_right;
+    for (size_t i = 0; i < sz * sz; i++)
+        m_right.push_back({distr(eng), distr(eng)});
+
+    auto m_right_tr = Pennylane::Util::Transpose(m_right, sz, sz);
+
+    for (auto _ : state) {
+        std::vector<std::complex<T>> m_out(sz * sz);
+
+        Pennylane::Util::omp_matrixMatProd(m_left.data(), m_right_tr.data(),
+                                           m_out.data(), sz, sz, sz,
+                                           Trans::Transpose);
+        benchmark::DoNotOptimize(m_out[sz * sz - 1]);
+    }
+}
+BENCHMARK(omp_matrixMatProd_cmplx<float>)
+    ->RangeMultiplier(1l << 2)
+    ->Range(1l << 4, 1l << 8);
+
+BENCHMARK(omp_matrixMatProd_cmplx<double>)
+    ->RangeMultiplier(1l << 2)
+    ->Range(1l << 4, 1l << 8);
+
+#if __has_include(<cblas.h>) && defined _ENABLE_BLAS
```

Comment:
This may be a tangent to this PR, but do we enable a build with these options enabled in CI. If not, we probably should.

Reply:
Thank you @mlxd for reviewing the PR! I added a new C++ tests with OpenBlas in CI (check `tests.yml`). Please, let me know if this addresses all the concerns.